core/server/proxy/src/main/java/alluxio/proxy/s3/S3RestServiceHandler.java

```diff
@@ -163,6 +164,22 @@ public S3RestServiceHandler(@Context ServletContext context) {
               () -> mAsyncAuditLogWriter != null
                   ? mAsyncAuditLogWriter.getAuditLogEntriesSize() : -1);
     }
+
+    // Initiate the S3 API metadata directories
+    if (!mMetaFS.exists(new AlluxioURI(S3RestUtils.MULTIPART_UPLOADS_METADATA_DIR))) {
```

Comment:
could you elaborate more on why we need to initiate the metadata directories since I don't have too much context about the change? 
Also it's not very clear to me why service handler constructor throw checked exception. I feel like it should be runtime exception if we can't initiate the handler.

Reply:
There is a custom Alluxio FS path `/.alluxio_s3_api_metadata` which is where I persist metadata for stateful API requests like multipart uploads.

I am using ACL ownership to denote which metadata belongs to which FS user, but need the  parent directory to be accessible by all users. To simplify the method calls, I'm initializing that directory here rather than handling it in every method which involves this directory.

Reply:
The explanation about the checked exception types is... exception handling is messy in the S3 API. The `S3Exception` is returned back to the S3 client as an XML, and so we need to wrap underlying exceptions into this class for that serialization to work.
- There's also some additional processing which translates `Exception`s into HTTP status codes & S3 error codes